platform/src/main/java/org/stellar/anchor/platform/service/Sep31DepositInfoGeneratorApi.java

```diff
@@ -0,0 +1,39 @@
+package org.stellar.anchor.platform.service;
+
+import java.util.Objects;
+import org.stellar.anchor.api.callback.GetUniqueAddressResponse;
+import org.stellar.anchor.api.callback.UniqueAddressIntegration;
+import org.stellar.anchor.api.exception.AnchorException;
+import org.stellar.anchor.api.sep.sep31.Sep31DepositInfo;
+import org.stellar.anchor.sep31.Sep31DepositInfoGenerator;
+import org.stellar.anchor.sep31.Sep31Transaction;
+import org.stellar.anchor.util.MemoHelper;
+
+public class Sep31DepositInfoGeneratorApi implements Sep31DepositInfoGenerator {
+  private UniqueAddressIntegration uniqueAddressIntegration;
+
+  public Sep31DepositInfoGeneratorApi(UniqueAddressIntegration uniqueAddressIntegration) {
+    this.uniqueAddressIntegration = uniqueAddressIntegration;
+  }
+
+  @Override
+  public Sep31DepositInfo generate(Sep31Transaction txn) throws AnchorException {
+    GetUniqueAddressResponse response = uniqueAddressIntegration.getUniqueAddress(txn.getId());
+    GetUniqueAddressResponse.UniqueAddress uniqueAddress = response.getUniqueAddress();
+    if (uniqueAddress.getMemo() == null) {
+      uniqueAddress.setMemo("");
+    }
+
+    if (uniqueAddress.getMemoType() == null) {
+      uniqueAddress.setMemoType("");
+    }
```

Comment:
Why are you setting an empty string when the memo and memoType are null? I think that if they're null we should keep the null value, right?

Or is this just a workaround for running `MemoHelper.makeMemo`?

Reply:
Removed these lines and updated the memo validation.

```java
    if (!Objects.toString(uniqueAddress.getMemo(), "").isEmpty()
        && !Objects.toString(uniqueAddress.getMemoType(), "").isEmpty()) {
      // Check the validity of the returned memo and memo type
      MemoHelper.makeMemo(uniqueAddress.getMemo(), uniqueAddress.getMemoType());
    }
```